### PR TITLE
Remove.wit.deployments.api.calls

### PR DESCRIPTION
**Pull request by corinnekrych**
Related to fixes https://openshift.io/openshiftio/Openshift_io/plan/detail/1315
- This PR makes sure all call to WIT deployments API are removed in the UI (before we could merge PR: https://github.com/fabric8-services/fabric8-wit/pull/2395).
- It does not change any call to a-runtime-console

### CODE REVIEW

**Comment by corinnekrych** [comment]
@christianvogt done.
If we don't want to ckeck-in packages/<project>/package-lock.json we should add them to .gitignore, wdyt?

**Comment by christianvogt** [comment]
Ya we should add exclusion to the packages for now. When I install & bootstrap I do not get the package-lock generated in the sub dir.

File: packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts [hunk 1]
```diff
@@ -39,9 +35,7 @@ export class AppLauncherDependencyCheckService implements DependencyCheckService
    *
    * @returns Observable
    */
-  getApplicationsInASpace(): Observable<Application[]> {
-    return this.deploymentApiService.getApplications(
-      this.context.space ? this.context.space.id : '',
-    );
+  getApplicationsInASpace(): Observable<any[]> {
+    return of([])
```
Comment (christianvogt) [inline packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @vikram-raj how is this used in launcher?

Comment (vikram-raj) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @christianvogt It is use to validate the application name while creating the application. So, that user can not create multiple application with the same name.

**Comment by corinnekrych** [comment]
@christianvogt @vikram-raj using `PipelinesService` with BuildConfig information gives us the name of the application available on OS, so the validation is added back on app creation flow:
unspecified

File: packages/fabric8-ui/src/app/space/add-app-overlay/add-app-overlay.component.spec.ts [hunk 2]
```diff
@@ -216,15 +212,6 @@ describe('AddAppOverlayComponent', () => {
       expect(component.applications).toEqual([]);
     });
 
-    it('should retieve applications if the current space is defined', () => {
```
Comment (christianvogt) [inline packages/fabric8-ui/src/app/space/add-app-overlay/add-app-overlay.component.spec.ts]: Can we keep these test cases now that you've found a work around to the list of applications.

File: packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts (hunk 1 above)
Comment (christianvogt) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @corinnekrych can we use the same workaround here to get the application names?

Comment (vikram-raj) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @corinnekrych ```getApplicationsInSpace``` is getting called in ```project-name.validator``` directive in ngx-launcher. https://github.com/fabric8-launcher/ngx-launcher/blob/master/projects/ngx-launcher/src/lib/shared/project-name.validator.ts#L34

**Comment by corinnekrych** [comment]
@christianvogt I put back the tests, unit tests are ok now but CI is failing with this:
```
FATAL: could not log in into vault
com.bettercloud.vault.VaultException: Vault responded with HTTP status code: 502
	at com.bettercloud.vault.api.Auth.loginByAppRole(Auth.java:386)
	at com.datapipe.jenkins.vault.credentials.VaultAppRoleCredential.authorizeWithVault(VaultAppRoleCredential.java:42)
Caused: com.datapipe.jenkins.vault.exception.VaultPluginException: could not log in into vault
	at com.datapipe.jenkins.vault.credentials.VaultAppRoleCredential.authorizeWithVault(VaultAppRoleCredential.java:44)
	at com.datapipe.jenkins.vault.VaultAccessor.auth(VaultAccessor.java:29)
```

🤔 #puzzled

**Comment by corinnekrych** [comment]
[test]

File: packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts (hunk 1 above)
Comment (corinnekrych) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @edewit do you know why we used project-name.validator given that we've already checked the name is not already in used in https://github.com/fabric8-ui/fabric8-ui/blob/8ebda0a75136aaaa525d54da5c018e026ce01409/packages/fabric8-ui/src/app/space/add-app-overlay/add-app-overlay.component.ts#L165 ?

Comment (christianvogt) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: @corinnekrych Looks like the app launcher has two more areas after the initial screen where the user can change the application name:

image
image

**Comment by christianvogt** [comment]
Something is wrong now with local dev and login. I'm getting a `401` error on load calling endpoint `/api/user/services` which is then deleting the `auth_token` from localstorage before the app loads which then puts me back to the login button screen.

I cannot login to the dev app.

**Comment by christianvogt** [comment]
Looked into the login issue a bit further. This is caused by the `add-app-overlay.component` now requiring the pipeline service which is run immediately even though the `add-app-overlay.component` isn't even shown but it is initialized.

@vikram-raj @rohitkrai03 could one of you help @corinnekrych in addressing this issue. Maybe somehow move the init code to only when the modal is actually needed rather than on app load.

**Comment by corinnekrych** [comment]
@christianvogt @rohitkrai03 indeed testing with a new incognito browser I can see the issue :( 
the injection of PipelinesService seems to be the root cause. 
I'm 👀  too.

**Comment by corinnekrych** [comment]
Interesting i've just discovered https://github.com/fabric8-ui/fabric8-ui/blob/master/packages/fabric8-ui/src/app/space/create/pipelines/services/pipelines.service.ts#L131

File: packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts (hunk 1 above)
Comment (corinnekrych) [reply packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-dependency-check.service.ts]: gotcha indeed!

**Comment by corinnekrych** [comment]
not to worry @edewit enjoy your ⛷ holidays, we got it sorted with, as Christian identified where it was used:
unspecified-3

@christianvogt ready for another review

**Comment by edewit** [comment]
Sorry but I have no idea

**Pull request merged**